src/ringmesh/io/geomodel_builder_resqml.cpp

```diff
@@ -0,0 +1,968 @@
+
+                    unit_2_info_[unit].layer_ = layer;
+                    if( unit->getInterpretedFeature() != nullptr )
+                    {
+                        const Layer3D& layer3d = static_cast< const Layer3D& >(
+                            geomodel_.geological_entity(
+                                GeologicalEntityType( "Layer" ),
+                                layer.index() ) );
+                        const_cast< Layer3D& >( layer3d ).set_name(
+                            unit->getInterpretedFeature()->getTitle() );
+                        unit_2_info_[unit].name_ =
+                            unit->getInterpretedFeature()->getTitle();
+                    }
+                }
+
+                for( auto contactIndex :
+                    range( rank_interp->getContactCount() ) )
+                {
+                    unit_2_info_[rank_interp->getSubjectOfContact(
+                                     contactIndex )]
+                        .relation_base_ = get_contact_mode(
+                        rank_interp->getSubjectContactModeOfContact(
+                            contactIndex ) );
+                    unit_2_info_[rank_interp->getSubjectOfContact(
+                                     contactIndex )]
+                        .interface_base_ = interp_2_geo_entity_
+                        [(AbstractFeatureInterpretation*)
+                                rank_interp->getHorizonInterpretationOfContact(
+                                    contactIndex )];
+
+                    unit_2_info_[rank_interp->getDirectObjectOfContact(
+                                     contactIndex )]
+                        .relation_top_ = get_contact_mode(
+                        rank_interp->getDirectObjectContactModeOfContact(
+                            contactIndex ) );
+                    unit_2_info_[rank_interp->getDirectObjectOfContact(
+                                     contactIndex )]
+                        .interface_top_ = interp_2_geo_entity_
+                        [(AbstractFeatureInterpretation*)
+                                rank_interp->getHorizonInterpretationOfContact(
+                                    contactIndex )];
+                }
+
+                // assign regions to layers
+                // TODO this is a workaround until the
+                // SealedVolumeFrameworkRepresentation
+                // is released in fesapi that tells us the layer information of
+                // a region
+                for( auto& region : geomodel_.regions() )
+                {
+                    gmge_id layer = find_layer( region );
+                    ringmesh_assert( layer.is_defined() );
+                    const Layer3D& layer3d = static_cast< const Layer3D& >(
+                        geomodel_.geological_entity(
+                            GeologicalEntityType( "Layer" ), layer.index() ) );
+                    const_cast< Region3D& >( region ).set_name(
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor the code to not need "const_cast". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS859)